From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nicolas Iooss <nicolas.iooss_linux@m4x.org>
Date: Sun, 19 Aug 2018 14:41:32 +0200
Subject: [PATCH] {MAYBE UPS} fs/cifs: do not use maybe-uninitialized
 resp_buftype

As rsp is NULL, this is not so important.
---
 fs/cifs/smb2pdu.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/fs/cifs/smb2pdu.c b/fs/cifs/smb2pdu.c
index 05149862aea4..843d8e7b8cb3 100644
--- a/fs/cifs/smb2pdu.c
+++ b/fs/cifs/smb2pdu.c
@@ -2645,7 +2645,8 @@ SMB2_open(const unsigned int xid, struct cifs_open_parms *oparms, __le16 *path,
 			    oparms->fid->lease_key, oplock, buf);
 creat_exit:
 	SMB2_open_free(&rqst);
-	free_rsp_buf(resp_buftype, rsp);
+	if (rsp)
+		free_rsp_buf(resp_buftype, rsp);
 	return rc;
 }
 
@@ -2964,7 +2965,8 @@ SMB2_close_flags(const unsigned int xid, struct cifs_tcon *tcon,
 
 close_exit:
 	SMB2_close_free(&rqst);
-	free_rsp_buf(resp_buftype, rsp);
+	if (rsp)
+		free_rsp_buf(resp_buftype, rsp);
 	return rc;
 }
 
@@ -3155,7 +3157,8 @@ query_info(const unsigned int xid, struct cifs_tcon *tcon,
 
 qinf_exit:
 	SMB2_query_info_free(&rqst);
-	free_rsp_buf(resp_buftype, rsp);
+	if (rsp)
+		free_rsp_buf(resp_buftype, rsp);
 	return rc;
 }
 
-- 
